jupyterhub/templates/proxy/service.yaml

```diff
@@ -45,12 +45,6 @@ spec:
     {{- if $HTTPS }}
     - name: https
       port: 443
-      # When HTTPS termination is handled outside our helm chart, pass traffic
-      # coming in via this Service's port 443 to targeted pod's port meant for
-      # HTTP traffic.
-      {{- if $offloadHTTPS }}
-      targetPort: http
-      {{- else }}
```

Comment:
This has always confused me greatly in the past, but I think it was added because when this port was open, it meant that `PROXY_SERVICE_PORT` environment variables would pick up on this, being the first in the list, which suddenly was unable to accept traffic if it went to the HTTPS port.

This PR will make it so that anyone sending traffic to `PROXY_PUBLIC_SERVICE_PORT` will suddenly get to the redirection. For those that know they want to speak HTTP without leaving the cluster etc would need to specify `PROXY_PUBLIC_SERVICE_HTTPS_PORT` environment variable. I'm not sure this is relevant, but I recall there have been issues for people with what the first port definition of the service is.